src/backend/distributed/connection/remote_commands.c

```diff
@@ -187,6 +187,35 @@ LogRemoteCommand(MultiConnection *connection, const char *command)
 
 /* wrappers around libpq functions, with command logging support */
 
+
+/*
+ * ExecuteCriticalRemoteCommand executes a remote command that is critical
+ * to the transaction. If the command fails then the transaction aborts.
+ */
```

Comment:
Not particularly happy about the name, but I can't think of something a lot better.